contrail-provisioner/analyticsnodes/analyticsnodes.go

```diff
@@ -0,0 +1,180 @@
+package analyticsnodes
+
+import (
+	"log"
+	"os"
+
+	contrailTypes "github.com/Juniper/contrail-operator/contrail-provisioner/contrail-go-types"
+	"github.com/Juniper/contrail-operator/contrail-provisioner/contrailclient"
+)
+
+// AnalyticsNode struct defines Contrail Analytics node
+type AnalyticsNode struct {
+	IPAddress   string            `yaml:"ipAddress,omitempty"`
+	Hostname    string            `yaml:"hostname,omitempty"`
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
+const analyticsNodeType string = "analytics-node"
+
+var analyticsInfoLog *log.Logger
+
+func init() {
+	analyticsInfoLog = log.New(os.Stdout, "analyticsnodes: ", log.LstdFlags)
+}
+
+// Create creates a AnalyticsNode instance
+func (c *AnalyticsNode) Create(nodeList []*AnalyticsNode, nodeName string, contrailClient contrailclient.ApiClient) error {
+	analyticsInfoLog.Println("Creating " + c.Hostname + " " + analyticsNodeType)
+	for _, node := range nodeList {
+		if node.Hostname == nodeName {
+			vncNode := &contrailTypes.AnalyticsNode{}
+			vncNode.SetFQName("", []string{"default-global-system-config", nodeName})
+			vncNode.SetAnalyticsNodeIpAddress(node.IPAddress)
+			annotations := contrailclient.ConvertMapToContrailKeyValuePairs(node.Annotations)
+			vncNode.SetAnnotations(&annotations)
+			err := contrailClient.Create(vncNode)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Update updates a AnalyticsNode instance
+func (c *AnalyticsNode) Update(nodeList []*AnalyticsNode, nodeName string, contrailClient contrailclient.ApiClient) error {
+	analyticsInfoLog.Println("Updating " + c.Hostname + " " + analyticsNodeType)
+	for _, node := range nodeList {
+		if node.Hostname == nodeName {
```

Comment:
This comparison is risky since Hostname might not always equal to nodeName.
Hostname may include capital letters but kubernetes object names doesn't.

Reply:
Now hostname is used everywhere